test/extensions/ERC721ABurnable.test.js

```diff
@@ -106,6 +112,22 @@ const createTestSuite = ({ contract, constructorArgs }) =>
         expect(await this.erc721aBurnable.balanceOf(this.addr1.address)).to.be.equal(this.numTestTokens - 1);
       });
 
+      it('startTimestamp updated correctly', async function () {
+        const tokenIdToBurn = this.burnedTokenId + 1;
+        const ownershipBefore = await this.erc721aBurnable.getOwnershipAt(tokenIdToBurn);
+        const timestampBefore = parseInt(ownershipBefore.startTimestamp);
+        const timestampToMine = (await getBlockTimestamp()) + 100;
+        await mineBlockTimestamp(timestampToMine);
+        const timestampMined = await getBlockTimestamp();
+        await this.erc721aBurnable.connect(this.addr1).burn(tokenIdToBurn);
+        const ownershipAfter = await this.erc721aBurnable.getOwnershipAt(tokenIdToBurn);
+        const timestampAfter = parseInt(ownershipAfter.startTimestamp);
+        expect(timestampBefore).to.be.lt(timestampToMine);
+        expect(timestampAfter).to.be.gte(timestampToMine);
+        expect(timestampAfter).to.be.gt(timestampBefore);
+        expect(timestampToMine).to.be.eq(timestampMined);
```

Comment:
Unlike Foundry, the next transaction after mining a timestamp may have a `block.timestamp` that is slightly higher than the mined timestamp.

Which is why we need these many checks.